src/Stryker.Core/Stryker.Core/Initialisation/InputFileResolver.cs

```diff
@@ -424,6 +461,19 @@ private IEnumerable<string> ExtractProjectFolders(ProjectAnalyzerResult projectA
             return folders;
         }
 
+        private IEnumerable<string> FindSharedProjects(XDocument document)
```

Comment:
Since this actually reads the project file, I don't think it can belong anywhere except in the ProjectFileReader unfortunately. I am confused as to why a new ProjectFileReader was created before the change as we have a project file reader already from the constructor. Perhaps the whole ExtractProjectFolders should be moved to the project file reader as well. Maybe rename the method to GetSharedProjectFolders or something.

Reply:
Actually it does read the file, only parses it. ProjectFileReader and InputFileResolver are due for refactoring but I have no idea on how to proceed at this stage.

My initial attempt was to improve the existing design and get rid of the `new ProjectFileReader()` and reuse the existing instance, but it ends up broking many tests and  also lead me to the realisation I had to replicate the `ExtractProjectFolders` logic for many of those to work.

Reply:
I suppose the refactor is too big to also take up in this PR, I have created a new issue for it.